engine/src/main/java/com/sandboni/core/engine/sta/analysis/ContextAnalyzer.java

```diff
@@ -0,0 +1,31 @@
+package com.sandboni.core.engine.sta.analysis;
+
+import com.sandboni.core.engine.filter.VertexFilterFactory;
+import com.sandboni.core.engine.sta.Context;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.sandboni.core.engine.filter.VertexFilterFactory.VertexFilterTypes.REFLECTION_SRC;
+
+public class ContextAnalyzer {
+
+    private ContextAnalyzer() {
+        throw new IllegalStateException("Utility class");
```

Comment:
who can call it..?  :)

Reply:
no one. the method is static. I'm making sure there is no unnecessary instantiation (sonar scan fix)